ValveKeyValue/ValveKeyValue/Deserialization/KV1BinaryReader.cs

```diff
@@ -61,82 +61,81 @@ public void Dispose()
 
         void ReadObjectCore()
         {
-            var type = ReadNextNodeType();
-            var name = ReadNullTerminatedString();
-            ReadValue(name, type);
-        }
-
-        void ReadValue(string name, KV1BinaryNodeType type)
-        {
-            KVValue value;
-
-            switch (type)
+            while (true)
```

Comment:
I need to check the larger structure, but looping here seems wrong. We're trying to read one object, not many. If we're just not reading `End` bytes then we should add that, as a peek if necessary.

Reply:
I've removed the loop and moved logic back to ReadObjectCore and ReadValue, which works fine.